src/kixi/datastore/schemastore/elasticsearch.clj

```diff
@@ -104,10 +106,16 @@
     component/Lifecycle
     (start [component]
       (if-not conn
-        (let [[host port] (if discover (es/discover-executor discover) [host port])
-              connection (es/connect host port)
+        (let [{:keys [native-host-ports http-host-ports]} (if discover
```

Comment:
I figure a repeat of this stuff is ok for clarity, but the rule of three should kick in if we have a third.